### PR TITLE
chore: add README table of contents

### PR DESCRIPTION
**Pull request by yftacherzog**
This PR introduces markers on the README file for table of contents to be used by an application to generate it, alongside an action that checks whether it is up to date or not. CI will prompt the user to run the command if the table of contents is out dated.

### CODE REVIEW

**Comment by yftacherzog** [comment]
Example error: https://github.com/konflux-ci/konflux-ci/actions/runs/9190224546/job/25273975445?pr=90

File: .github/workflows/check-toc.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Verify ToC
+        run: |
+          npx markdown-toc README.md -i
+          if [[ $(git status --porcelain) ]]; then
+              echo "Did you forget to update the README table of contents?" >&2
+              echo "Please run 'npx markdown-toc README.md -i' and update your PR" >&2
```
Comment (avi-biton) [inline .github/workflows/check-toc.yaml]:
```suggestion
              echo "Error: Did you forget to update the README table of contents?" >&2
              echo "Error: Please run 'npx markdown-toc README.md -i' and update your PR" >&2
```
I think that in order to make it easy to understand where the error is, we should add `Error` before the message.

Comment (yftacherzog) [reply .github/workflows/check-toc.yaml]: done

**Review by avi-biton — APPROVED** [review]
lgtm

**Pull request merged**